### DISCUSSION TITLE
AO3's built-in download option?

### DISCUSSION THREAD

**Discussion by Kyther** [nianeyna/ao3downloader#93] [Q&A] [answered]
I tried looking at the code but I barely know anything about Python so I couldn't tell:  Does this use AO3's download methods or does it scrape the work and process it in some other fashion?  I'm assuming it uses the built-in download option, but wanted to check.

**Comment by nianeyna** [comment, marked as answer]
It does indeed use the built-in download buttons! 

If you happen to not prefer that for whatever reason, I know some people have had success using the "get links only" option of this program and giving the link list generated by it to a [Calibre](https://calibre-ebook.com/) extension called [FanFicFare](https://github.com/JimmXinu/FanFicFare), which I believe uses its own download format (unfortunately I can't give you any more details on that, as I don't use FanFicFare). 

If you do use FanFicFare, the reason to generate the links list using ao3downloader instead of giving the page url directly to FanFicFare itself is that ao3downloader will get all links from all pages of the url you give it, whereas FanFicFare in isolation requires you to enter each page link individually.

That's probably much more information that you needed, lol.

**Reply by Kyther** [reply]
Ha, I like lots of information.  I don't use Calibre for anything but converting when absolutely necessary, so I don't use FanFicFare, but I have friends who do.

I once used a scraper on here (by radiolarian I think?) that would get all the work links, but it didn't have authentication offered, and especially these days with so many people locking work to logged-in users only thanks to the AI scraping, it's much more common to run across archive-locked fics.
